### ISSUE TITLE
Why pretrained a word2vec word embedding  only on the train set?

### ISSUE THREAD

**Issue by QuietWoods** [ChenRocks/fast_abs_rl#13]
https://github.com/ChenRocks/fast_abs_rl/blob/aebf539107caba5be35720f5d1f9f98989a069e8/train_word2vec.py#L24
Excuse me, I found that you only pretrained a word2vec word embedding on the train set，not included val set. Have any deep meaning? I am confused about this.Thanks！

**Comment by ShengleiH** [comment]
@QuietWoods May be this answer could solve your questions about pertained word2vec only on train dataset: 
[https://github.com/guillaumegenthial/sequence_tagging/issues/46#issuecomment-382718474](url)

In conclusion, I think this is because the author here trained a new word2vec on train set, instead of using a pretrained word2vec like 'glove' etc. If you use pretrained word2vec like 'glove' and freeze the word embedding, then it is okey to include val and test set's vocabulary, since the training of 'glove' does not involve the data used in this task. But if you train a new word embedding on the data  involved in this task, then it is better not to include val and test set's vocabulary.

Maybe I have made some mistakes, welcome to discuss with me :)

**Comment by Stephen-Adams** [comment]
> [fast_abs_rl/train_word2vec.py](https://github.com/ChenRocks/fast_abs_rl/blob/aebf539107caba5be35720f5d1f9f98989a069e8/train_word2vec.py#L24)
> 
> Line 24 in [aebf539](/ChenRocks/fast_abs_rl/commit/aebf539107caba5be35720f5d1f9f98989a069e8)
> 
>  self._path = join(DATA_DIR, 'train') 
> 
> Excuse me, I found that you only pretrained a word2vec word embedding on the train set，not included val set. Have any deep meaning? I am confused about this.Thanks！

hello, I follow the step in https://github.com/ChenRocks/cnn-dailymail to preprocess the data, and get train.tar, val.tar and test.tar. And I don't know what is the file "join(DATA_DIR, 'train')" used to pretrain word2vec. Could you please help me?

**Comment by Stephen-Adams** [comment]
> [fast_abs_rl/train_word2vec.py](https://github.com/ChenRocks/fast_abs_rl/blob/aebf539107caba5be35720f5d1f9f98989a069e8/train_word2vec.py#L24)
> 
> Line 24 in [aebf539](/ChenRocks/fast_abs_rl/commit/aebf539107caba5be35720f5d1f9f98989a069e8)
> 
>  self._path = join(DATA_DIR, 'train') 
> 
> Excuse me, I found that you only pretrained a word2vec word embedding on the train set，not included val set. Have any deep meaning? I am confused about this.Thanks！

I tried file train.tar, but it seems a directory is needed.
